### PR TITLE
Add liquibase to project

### LINKED ISSUE #36: Add liquibase to project

### PR DESCRIPTION
**Pull request by tstepnik**
Add first initial schema with current DB. I'm not sure if constraints name is everywere correct.
Closes #36

### CODE REVIEW

File: src/main/java/com/tstepnik/client/main.css [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#header{
```
Comment (rbobko93) [inline src/main/java/com/tstepnik/client/main.css]: what is this file?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -5,4 +5,4 @@ spring.datasource.password=
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 spring.jpa.hibernate.ddl-auto=update
 spring.h2.console.enabled=true
-
+spring.liquibase.change-log=classpath:liquibase/master.xml
```
Comment (rbobko93) [inline src/main/resources/application.properties:7]: 👍

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        <createTable tableName="user_role">
+            <column name="id" type="bigint" autoIncrement="${autoIncrement}">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="role" type="varchar(255)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+
+        <createTable tableName="user_roles">
```
Comment (rbobko93) [inline src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:43]: Does this work? Usually name of "many to many" table is concatenation of two table names: `user_user_role`

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                                 referencedTableName="user"
+                                 referencedColumnNames="id"/>
+
+        <addForeignKeyConstraint baseTableName="user_roles"
+                                 baseColumnNames="roles_id"
+                                 constraintName="fk_role"
+                                 referencedTableName="user_role"
+                                 referencedColumnNames="id"/>
+
+        <createTable tableName="task">
```
Comment (rbobko93) [inline src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:64]: Did you add all the fields for the task? What about planned for and creation date?

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            <column name="id" type="bigint" autoIncrement="${autoIncrement}">
+                <constraints nullable="false" primaryKey="true"/>
+            </column>
+            <column name="description" type="varchar(1000)">
```
Comment (rbobko93) [inline src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:68]: I am not sure if you can set size to a 1000. We talked about storing large strings.

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+                <constraints nullable="false"/>
+            </column>
+            <column name="importance" type="varchar(255)">
```
Comment (rbobko93) [inline src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:71]: importance is never null

File: src/main/resources/liquibase/master.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext
+   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd
+   http://www.liquibase.org/xml/ns/dbchangelog
+   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">
+
```
Comment (rbobko93) [inline src/main/resources/liquibase/master.xml:9]: you have to add initial_schema.xml here because right now it is never run. Did you test if it works? I'm pretty sure that hibernate is still creating your schema.

**Review by rbobko93 — CHANGES_REQUESTED** [review]

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml (hunk 4 above)
Comment (tstepnik) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:64]: They are not in develop branch so I didn't.

File: src/main/java/com/tstepnik/client/main.css (hunk 1 above)
Comment (tstepnik) [reply src/main/java/com/tstepnik/client/main.css]: Heh I have been learning css and forgot removed it.

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml (hunk 3 above)
Comment (tstepnik) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:43]: I did it according to schema which you sent me from github.

Comment (tstepnik) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:43]: Before liquibase hibernate  set this name for third table so I just copied the name.

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml (hunk 5 above)
Comment (tstepnik) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:68]: I checked if I can and field is properly added to DB.

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml (hunk 3 above)
Comment (rbobko93) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:43]: I am pretty sure that hibernate named this table just like I suggested above. Usually it is the name of two tables concatenated together.

File: src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml (hunk 5 above)
Comment (rbobko93) [reply src/main/resources/liquibase/changelog/00000000000000_initial_schema.xml:68]: coolio

File: src/main/resources/liquibase/master.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    <include file="changelog/00000000000000_initial_schema.xml" relativeToChangelogFile="true"/>
```
Comment (rbobko93) [inline src/main/resources/liquibase/master.xml:10]: 👍

**Review by rbobko93 — COMMENTED** [review]
One small comment :)

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1,8 +1,8 @@
-spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.url=jdbc:h2:mem:testdb;
 spring.datasource.driverClassName=org.h2.Driver
 spring.datasource.username=root
 spring.datasource.password=
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
 spring.jpa.hibernate.ddl-auto=update
```
Comment (rbobko93) [inline src/main/resources/application.properties]: I think we need to drop this guy since we are using liquibase now

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -1,8 +1,8 @@
-spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.url=jdbc:h2:mem:testdb;
```
Comment (rbobko93) [inline src/main/resources/application.properties]: Why did you add colon (;) here?

Comment (tstepnik) [reply src/main/resources/application.properties]: probably do some changes and missed it.

**Comment by rbobko93** [comment]
@tstepnik use better commit messages. Simple "remove" doesn't work. Imagine you coming back to this code year leater. You would have to look into each commit to figure out what it did because message doesn't say anything about it.

**Review by rbobko93 — APPROVED** [review]
